pkg/blobinfocache/boltdb/boltdb.go

```diff
@@ -325,7 +325,67 @@ func (bdc *cache) appendReplacementCandidates(candidates []prioritize.CandidateW
 // data from previous RecordDigestUncompressedPair calls is used to also look up variants of the blob which have the same
 // uncompressed digest.
 func (bdc *cache) CandidateLocations2(transport types.ImageTransport, scope types.BICTransportScope, primaryDigest digest.Digest, canSubstitute bool) []blobinfocache.BICReplacementCandidate2 {
-	return bdc.candidateLocations(transport, scope, primaryDigest, canSubstitute, true)
+	resAllBlobs := []prioritize.CandidateWithTime{}
+	var uncompressedDigestValue digest.Digest // = ""
+	requireCompressionInfo := true
+	resultPrioritized := bdc.candidateLocations(transport, scope, primaryDigest, canSubstitute, requireCompressionInfo)
+	// Reprocess all blobs and append them to resultPrioritized in lower priority
+	if err := bdc.view(func(tx *bolt.Tx) error {
+		scopeBucket := tx.Bucket(knownLocationsBucket)
+		if scopeBucket == nil {
+			return nil
+		}
+		scopeBucket = scopeBucket.Bucket([]byte(transport.Name()))
+		if scopeBucket == nil {
+			return nil
+		}
+		if err := scopeBucket.ForEach(func(name []byte, v []byte) error {
+			bucket := scopeBucket.Bucket(name)
+			// compressionBucket won't have been created if previous writers never recorded info about compression,
+			// and we don't want to fail just because of that
+			compressionBucket := tx.Bucket(digestCompressorBucket)
+
+			resAllBlobs = append(resAllBlobs, bdc.appendReplacementCandidates(resAllBlobs, bucket, compressionBucket, primaryDigest, requireCompressionInfo)...)
+			if canSubstitute {
+				if uncompressedDigestValue = bdc.uncompressedDigest(tx, primaryDigest); uncompressedDigestValue != "" {
+					b := tx.Bucket(digestByUncompressedBucket)
+					if b != nil {
+						b = b.Bucket([]byte(uncompressedDigestValue.String()))
+						if b != nil {
+							if err := b.ForEach(func(k, _ []byte) error {
+								d, err := digest.Parse(string(k))
+								if err != nil {
+									return err
+								}
```

Comment:
Even without the `scopeBucket.ForEach`, doing this again outside of `.candidateLocations` seems unnecessary.

Reply:
Similar being disucssed above.